### DISCUSSION TITLE
Parse /// comments for @@ annotations

### DISCUSSION THREAD

**Discussion by m1212e** [prisma/prisma#20899] [Q&A]
### Question

Hi, did you consider parsing /// comments for @@ annotations?

See this issue for detailed context: https://github.com/Brakebein/prisma-generator-nestjs-dto/issues/28

I use a generator that outputs DTOs based on the schema. In concept, this happens:

The parser sees this model
```
model Role {
  id            String @id
  name      String
  attribute String

  @@unique([name, attribute])
}
```
And generates this
```
export class FindRoleDTO {
  id?: string;
  name_attribute?: RoleNameAttributeUniqueInputDto; // this is basically { name: string, attribute: string }
}
```
The generator allows ignoring certain fields you don't want to appear in the output by adding a comment like `/// @DTOHide` to a field. Now if I would like to disable output of the `name_attribute` I can't because the parser does not pick up the comment for the @@ annotation.

**Comment by ludralph** [comment]
Hi @m1212e 👋 

Thank you for raising this question.

This looks like a feature request as we support this functionality at the moment. Can you please open a [feature reuest](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=)  so that our engineers can take a look.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by janpio** [comment]
Are you reporting that `///` are not attached to information from the unique that is put into the DMMF that the generator is provided? So the `@@unique` is actually picked up, but the comment in front of it is not?

We have an issue around DMMF related problems here: https://github.com/prisma/prisma/issues/18561 But I do not see that mentioned there yet.

**Reply by m1212e** [reply]
Yes, thats exactly what I was referring to. Tbh I'm not super deep into the specifics of the parser but as far as I see what I mentioned is currently not possible. Would be cool to see this!

**Reply by ludralph** [reply]
Hi @m1212e 

Can you please document your use case in the open [issue](https://github.com/prisma/prisma/issues/18561)? This will help our engineers when prioritizing it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
